compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -404,6 +405,21 @@ generateSrcFromLf env = noLoc mod
                     , cid_datafam_insts = []
                     , cid_overlap_mode = Nothing
                     }
+      where
+        -- Split a DefValue's name, into the lexical part and the numeric part
+        -- if it exists. For example, the name "$dFooBar123" is split into a
+        -- pair ("$dFooBar", Just 123), and the name "$dFooBar" would be turned
+        -- into ("$dFooBar", Nothing). This gives us the correct (or, close
+        -- enough) order for recreating dictionary function names in GHC without
+        -- mismatches.
+        --
+        -- See issue #7362, and the corresponding regression test in the
+        -- packaging test suite.
+        nameKey :: LF.DefValue -> (T.Text, Maybe Int)
+        nameKey dval =
+            let name = LF.unExprValName (fst (LF.dvalBinder dval))
+                (intR,tagR) = T.span isDigit (T.reverse name)
+            in (T.reverse tagR, readMay (T.unpack (T.reverse intR)))
```

Comment:
It's a bit sad that `Data.Text` doesn't have `spanEnd`, unlike `Data.Bytestring`. 😿